src/states/virginia/main.py

```diff
@@ -12,30 +15,40 @@ def get_locality_ids():
   return [option['value'] for option in soup.select('select>option') if option['value']]
 
 
-def get_locality_datum(id_):
+def get_locality_datum(id_, attempt=0):
+  if attempt > 5:
+    return {}
   page = cache_request(
     BASE_URL,
     method='POST',
     data={'LocalityUid': id_},
-    wait=random.uniform(.1, .3),
+    wait=random.uniform(2, 3),
   )
   soup = BeautifulSoup(page, 'lxml')
-  keys = soup.select('.resultsWrapper')[0].select('h5.display-lable')
-  vals = soup.select('.resultsWrapper')[0].select('p.display-field')
-  results = {key.text.strip(): val.text.strip() for key, val in zip(keys, vals)}
-  locale = soup.select('select > option[selected="selected"]')[0].text.title()
-  final = {
-    'locale': locale,
-    'county': locale if locale.endswith('County') else None,
-    'city': locale if not locale.endswith('County') else None,
-    'official': results['Registrar'],
-    'emails': [results['Email']],
-    'faxes': [results['Fax']],
-    'url': results.get('URL'),
-    'address': results.get('Mailing Address') or results.get('Address'),
-    'physicalAddress': results.get('Physical Address'),
-  }
-  return {k: v for k, v in final.items() if v}
+  try:
+    keys = soup.select('.resultsWrapper')[0].select('h5.display-lable')
+    vals = soup.select('.resultsWrapper')[0].select('p.display-field')
+    results = {key.text.strip(): val.text.strip() for key, val in zip(keys, vals)}
+    locale = soup.select('select > option[selected="selected"]')[0].text.title()
+    final = {
+      'locale': locale,
+      'county': locale if locale.endswith('County') else None,
+      'city': locale if not locale.endswith('County') else None,
+      'official': results['Registrar'],
+      'emails': [results['Email']],
+      'faxes': [results['Fax']],
+      'url': results.get('URL'),
+      'address': results.get('Mailing Address') or results.get('Address'),
+      'physicalAddress': results.get('Physical Address'),
+    }
+    return {k: v for k, v in final.items() if v}
+  except IndexError:
```

Comment:
Instead of using a general `except`, check `if soup.select('.resultsWrapper')` instead